src/Detector/DotNetCore/ProjectFileProviders/ProbeAndFindProjectFileProvider.cs

```diff
@@ -57,72 +57,85 @@ public string GetRelativePathToProjectFile(DetectorContext context)
-            bool webAppProjectExists = false;
-            bool blazorWasmProjectExists = false;
 
             foreach (var file in projectFiles)
             {
                 allProjects.Add(file);
                 if (ProjectFileHelpers.IsAzureBlazorWebAssemblyProject(sourceRepo, file))
                 {
-                    azureBlazorWasmProjects.Add(file);
-                    blazorWasmProjectExists = true;
+                    blazorWasmProjects.Add(file);
                 }
                 else if (ProjectFileHelpers.IsAzureFunctionsProject(sourceRepo, file))
                 {
                     azureFunctionsProjects.Add(file);
-                    functionProjectExists = true;
                 }
                 else if (ProjectFileHelpers.IsAspNetCoreWebApplicationProject(sourceRepo, file))
                 {
                     webAppProjects.Add(file);
-                    webAppProjectExists = true;
                 }
             }
 
             // Assumption: some build option "--apptype" will be passed to oryx
             // which will indicate if the app is an azure function type or static type app
 
-            // If there are multiple projects, we will look for --appty to detect corresponding project.
+            // If there are multiple projects, we will look for --apptype to detect corresponding project.
             // for example azurefunction and blazor both projects can reside
             // at the same repo, so more than 2 csprojs will be found. Now we will
             // look for --apptype value to determine which project needs to be built
             if (!string.IsNullOrEmpty(_options.AppType))
             {
                 _logger.LogInformation($"{nameof(_options.AppType)} is set to {_options.AppType}");
 
-                if (functionProjectExists && _options.AppType.ToLower().Contains(Constants.FunctionApplications))
+                var appType = _options.AppType.ToLower();
+                if (appType.Contains(Constants.FunctionApplications))
                 {
+                    if (azureFunctionsProjects.Count() == 0)
+                    {
+                        return null;
+                    }
+
                     projectFile = GetProject(azureFunctionsProjects);
                 }
-                else if (blazorWasmProjectExists
-                    && _options.AppType.ToLower().Contains(Constants.StaticSiteApplications))
+                else if (appType.Contains(Constants.StaticSiteApplications))
+                {
+                    if (blazorWasmProjects.Count() == 0)
+                    {
+                        return null;
+                    }
+
+                    projectFile = GetProject(blazorWasmProjects);
+                }
+                else if (appType.Contains(Constants.WebApplications))
                 {
-                    projectFile = GetProject(azureBlazorWasmProjects);
+                    if (webAppProjects.Count() == 0)
```

Comment:
why are we checking the count? so are you saying there is a possibility that apptype will be wrongly set ? like there is no webappproject but apptype is set to webapp

Reply:
Exactly, if the users are saying to look for a particular apptype we are looking only for those and if there are none we should not be returning other app type based projects